x/auth/types/credentials.go

```diff
@@ -32,25 +32,20 @@ const ModuleCredentialType = "ModuleCredential"
 
 var _ cryptotypes.PubKey = &ModuleCredential{}
 
+// NewModuleCredential creates new module credential key.
+// At least one derivation key must be provided. Panics otherwise.
 func NewModuleCredential(moduleName string, derivationKeys [][]byte) *ModuleCredential {
+	if len(derivationKeys) == 0 {
```

Comment:
Can we avoid a panic here and instead return an error? The caller should be responsible for dealing with the error.

Reply:
yes, but then it is API breaking.

Reply:
This is fine because this function has not been released. No this is not really API breaking.

Reply:
changed `NewModuleCredential` to: return error, and take variadic arguments (instead of a list)